llvm/include/llvm/Analysis/MemoryRefInfo.h

```diff
@@ -0,0 +1,57 @@
+//===--------- Definition of the MemoryRefInfo class -*- C++ -*------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines MemoryRefInfo class that is used when getting
+// the information of a memory reference instruction.
```

Comment:
Most of the patch is NFC which renames  this class,
Can you please split "move and rename" into one patch, and functional change into another.

Reply:
do you still need this patch?